src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,101 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_LENGTH = 10;
+    private int position = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[ARRAY_LENGTH];
+    }
+
+    @Override
+    public void add(T value) {
+        if (position >= array.length) {
+            int newArrayLength = array.length + (array.length >> 1);
+            array = Arrays.copyOf(array, newArrayLength);
+        }
+        array[position] = value;
+        position++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (position >= array.length || index > position) {
+            throw new IndexOutOfBoundsException();
+        } else {
+            T[] tempArray = (T[]) new Object[array.length + 1];
+            System.arraycopy(array, 0, tempArray, 0, index);
+            tempArray[index] = value;
+            System.arraycopy(array, index, tempArray, index + 1, array.length - index);
+            array = tempArray;
+            position++;
+        }
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        if (!list.isEmpty()) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        try {
+            return array[index];
```

Comment:
list = new ArrayList();
list.get(3);
test what happens here?

Reply:
will be return null, but must throws IndexOutOfBoundsException